Gdr 2419

gladkia opened the pull request:
# Description
Improve pkgdown site
## What changed?
- updated news and added reference sections
- Related JIRA issue:  GDR-2419
- pkgdown site link: https://gdrplatform.github.io/gDRcore/

# Logistic checklist
- [X] Package version bumped
- [X] Changelog updated

# Screenshots (optional)

j-smola commented on NEWS.md:
```diff
@@ -1,532 +1,537 @@
-## 0.1.3.3 (2022-09-27)
-- update the logic for parallel computing
+## gDRcore 0.1.3.4 - 2022-09-29
+* Change the logic for using cores in `BiocParallel`
```
```suggestion
* change the logic for using cores in `BiocParallel`
```

j-smola commented on NEWS.md:
```diff
@@ -1,532 +1,537 @@
-## 0.1.3.10 (2022-12-15)
-- add assert for vehicle values in input data in `runDrugResponseProcessingPipeline`
+## gDRcore 0.1.3.11 - 2022-12-16
+* Replace parallelize function with gDRutils::loop
```
```suggestion
* replace parallelize function with gDRutils::loop
```

j-smola commented on NEWS.md:
```diff
@@ -1,532 +1,537 @@
-## 0.1.3.12 (2023-01-12)
-- refactor mapping function to properly handle drug3
+## gDRcore 0.1.3.13 - 2023-01-13
+* Clean-up code
```
```suggestion
* clean-up code
```

j-smola commented on NEWS.md:
```diff
@@ -1,532 +1,537 @@
-## 0.1.3.13 (2023-01-13)
-- Clean-up code
+## gDRcore 0.1.3.14 - 2023-02-10
+* Clean-up code
```
```suggestion
* clean-up code
```

j-smola commented on NEWS.md:
```diff
@@ -1,532 +1,537 @@
-## 0.99.18 (2023-05-31)
-- fix managing of mixed types of raw data
+## gDRcore 0.99.19 - 2023-06-06
+* replaced reshape2 functions by functions from data.table
```
```suggestion
* replace reshape2 functions by functions from data.table
```

j-smola commented on NEWS.md:
```diff
@@ -1,532 +1,537 @@
-## 0.99.31 (2023-07-19)
-- update the logic for handling warnings in the pipeline
+## gDRcore 0.99.32 - 2023-07-25
+* extended logic for supporting cols with dash, e.g. additional perturbations with "-"
```
```suggestion
* extend logic for supporting cols with dash, e.g. additional perturbations with "-"
```

j-smola reviewed (commented):
Could you maybe rethink group for functions?
1. IMO not all function in "Combination helpers" is used only for combination data.
2. "Mappping" seems to be list of function supporting "Pipeline"
3. "Utils" and "Preparing input" are both "Preparing input for pipline"? As also "Identifying keys", right?
4. etc.



Please, consider also
1. update or "gDRcore.Rmd" vignette - "Drug processing section" section where due to lack of empty line list of steps disappeared;
5. update the gDRcore architecture diagram in READMA.md as it is no longer readable on the pkgdown site.

PS: And fot is the reasone for function from `constant.R` file?

gladkia commented:
> Could you maybe rethink group for functions?
> 
> 1. IMO not all function in "Combination helpers" is used only for combination data.
> 2. "Mappping" seems to be list of function supporting "Pipeline"
> 3. "Utils" and "Preparing input" are both "Preparing input for pipline"? As also "Identifying keys", right?
> 4. etc.
> 
> Please, consider also
> 
> 1. update or "gDRcore.Rmd" vignette - "Drug processing section" section where due to lack of empty line list of steps disappeared;
> 2. update the gDRcore architecture diagram in READMA.md as it is no longer readable on the pkgdown site.
> 
> PS: And fot is the reasone for function from `constant.R` file?

# 1st section
1. Thanks. Fixed.
2. Yes. Please note that `runDrugResponseProcessingPipeline()` is the top-level function for running the whole pipeline, dependent on multiple functions from gDRcore.
3. `Prepare input` refers to the S3 methods called 'prepare_input' for data.table and MAE as well as their dependencies (internal helper functions). The `utils` comprises a broader, less specific category  I've moved a single function from 'Identifying keys' to the utils category. 

# 2nd section
1. Done.
2. Good idea. I'll try to prepare an analogous diagram with plantuml.
3. 'constants.R' - there are no functions there - only three `constant` objects: `drugNameRegex`, `untreated_tag_patterns`, and `untreatedDrugNameRegex`.

j-smola commented on README.md:
```diff
@@ -1,58 +1,7 @@
-## Architecture
+## gDRcore
```
```suggestion
# gDRcore
```

j-smola commented on vignettes/gDRcore.Rmd:
```diff
@@ -39,14 +39,21 @@ a series or the same number of elements, the attributes (columns of the table) s
 ## Drug processing 
 For drug response data, the input files need to be merged such that each measurement (data) is associated with the right metadata (cell line properties and treatment definition). Metadata can be added with the function `cleanup_metadata` if the right reference databases are in place.
 
-When the data and metadata are merged into a long table, the wrapper function `runDrugResponseProcessingPipeline` can be used to generate an MAE with processed and analyzed data. In practice runDrugResponseProcessingPipeline does the following steps:
```
Maybe also this chunk wit loading libraries could be with `echo=FALSE`?

gladkia replied:
done.
